### ISSUE TITLE
Is convert timezone working as expected on bigquery?

### ISSUE THREAD

**Issue by zack-bodinger** [calogica/dbt-date#89]
Maybe I'm misunderstanding the intended functionality of the macro -- but I noticed 2 things for the bigquery implementation of this macro:
1. there is no option to read in the source_tz -- is that intentional?
2. more of interest and confusingly to me, it seems that the macro converts the provided column to the correct target_tz as a datetime, but in casting it back to a timestamp, it displays the converted timezone as UTC, which seems to me to be misleading. 

In the below screenshot see a few examples that have tried to convert the column on the left to PST using this macro:
<img width="486" alt="image" src="https://user-images.githubusercontent.com/22088864/201217628-79e8df03-95a1-4a48-ba5f-e4c3db5fe619.png">

Here, although the hours and minutes are correct, the timezone is displayed as UTC in the field on the right since this field has been converted back to a timestamp.... keeping it as a datetime would solve this problem. Could someone please clarify this behavior?

**Comment by clausherther** [comment]
@zack-bodinger timezone conversion across platforms is....confusing! Couple of thoughts:
- Last I checked, BigQuery does not allow you to specify a `source_tz` in its conversion [function](https://cloud.google.com/bigquery/docs/reference/standard-sql/datetime_functions#datetime), so we set that parameter to `None` in that version
- BigQuery also displays all timestamps with the UTC suffix in its UI, even if you've successfully converted it to PST or EST. Super confusing, but as far as I know only a UI issue. We raised this at a client site with Google a couple of years ago, but apparently no progress yet on this
- Timezone conversion logic is coming to dbt-core (see: https://github.com/dbt-labs/dbt-core/issues/5969), so once that's implemented, we're going to deprecate that functionality here and just use the core implementation, similar to how we've deprecated the dbt-utils date math functions recently.
